Add dialogs asking for permission, and add shizuku method for granting permission

TymanWasTaken opened the pull request:
- Push unfinished code
- It finally works (mostly)
- Push current broken code
- Push finished and functional code

TymanWasTaken committed:
Push unfinished code

TymanWasTaken committed:
It finally works (mostly)

TymanWasTaken committed:
Push current broken code

TymanWasTaken committed:
Push finished and functional code

TymanWasTaken commented:
Wait I forgot to change sdk level back

wingio commented:
> Wait I forgot to change sdk level back
> 

Keep it there, some other stuff ill use in the future are also a7+

wingio commented on app/src/main/java/xyz/wingio/logra/utils/Shizuku.kt:
```diff
@@ -0,0 +1,60 @@
+        // Set a listener for the permission grant/deny
+        Shizuku.addRequestPermissionResultListener { requestCode, grantResult ->
+            if (requestCode != shizukuRequestCode) return@addRequestPermissionResultListener
+            Logger("dn").debug(grantResult.toString())
```
guh???

TymanWasTaken committed:
Hanful of fixes

- Change sdk level back to 21, and check for < 23 before using shizuku

- Restart app instad of closing for shizuku

- Only use su once for root method

- Make root method manually restart as it doesn't get closed by the OS

TymanWasTaken committed:
Do not attempt to restart on root grant, instead just reconnect logcat manager

TymanWasTaken committed:
Remove grant and close dialog for root, as root does not close the app after granting
